docs/Configuration.md

```diff
@@ -15,6 +15,19 @@ const kafka = new Kafka({
 })
 ```
 
+## Client Id
+The [kafka documentation](https://kafka.apache.org/documentation/#design_quotasgroups) describes the `clientId` as:
+
+> Client-id is a logical grouping of clients with a meaningful name chosen by the client application. The tuple (user, client-id) defines a secure logical group of clients that share both user principal and client-id. Quotas can be applied to (user, client-id), user or client-id groups.
+
+It [also says](https://kafka.apache.org/documentation/#producerconfigs_client.id):
+
+> `client.id`
+>
+> An id string to pass to the server when making requests. The purpose of this is to be able to track the source of requests beyond just ip/port by allowing a logical application name to be included in server-side request logging.
+
+Therefore the `clientId` should be shared across multiple instances in a cluster or horizontally scaled application, but distinct for each application. Consumer group instances will also have [`clientHost`](../types/index.d.ts#L715) and [`memberId`](../types/index.d.ts#L717) fields which can be used to distinguish between instances with the same [`clientId`](../types/index.d.ts#L715).
```

Comment:
Those links won't resolve, since they are relative links to files in the repo, not to pages on the website.

Either way, I would remove this part, since it's only relevant for people operating Kafka clusters, not users of KafkaJS. If you're operating a Kafka cluster, you already know that you identify logical applications by clientId+principal, and individual instances by `memberId` or ip.

Our users just need to know what a good value to set is, so maybe try to condense this a bit. Something like "A logical identifier of an application. Can be used by brokers to apply quotas or trace requests to a specific application. Example: `booking-events-processor`". Short and sweet 😃

Reply:
Ok, sounds good. I added your phrase to the top verbatim and I left in the first sentence of this paragraph as that seems to be the sentence I was wishing was there. I think the reader can figure it out by reading and comprehending the quoted passages but it should help to spell it out plainly.